### PR TITLE
Issue #3217 new db size report option in schema tool

### PR DESCRIPTION
**Pull request by punktilious**

### CODE REVIEW

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/ISizeReport.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package com.ibm.fhir.schema.size;
+
+/**
+ * Render a size report
+ */
+public interface ISizeReport {
+
+    /**
+     * Render a report using the data from the given model
+     * @param os
```
Comment (tbieste) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/ISizeReport.java:16]: Fix param to match actual param.

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/PostgresSizeCollector.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                        logger.fine(String.format("%56s %34s %8d %10d %10d %10d", tableName, resourceType, rowEstimate, totalBytes, indexBytes, toastBytes));
+                    }
+                    model.accumulateTableSize(resourceType, tableName, isParamTable, totalBytes - indexBytes, rowEstimate);
+                }
+            }
+        } catch (SQLException x) {
+            throw translator.translate(x);
+        }
+    }
+    private void collectIndexInfo(String schemaName, Connection connection, IDatabaseTranslator translator) {
```
Comment (tbieste) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/PostgresSizeCollector.java:88]: Add a newline between methods.

File: fhir-persistence-schema/docs/SchemaToolUsageGuide.md [hunk 3]
```diff
@@ -522,14 +522,43 @@ Note: the jar file is stored locally in `fhir-persistence-schema/target` or in t
...
+```
+
+The detail rows are tab-separated, making it easy to load the data into a spreadsheet for further analysis.
+
+**Notes:**
+1. The size report is only supported on Db2 and PostgreSQL databases.
+2. The size report is intended as a guide to understand the relative space distribution of objects in the IBM FHIR Server data schema. The report is not intended to replace database utilities for calculating the total size of the database.
+
+----------------------------------------------------------------
+# List of IBM FHIR Server Persistence Schema Tool Flags
```
Comment (lmsurpre) [inline fhir-persistence-schema/docs/SchemaToolUsageGuide.md:561]: should we add the `--tenant-name` flag to this list as well?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/FHIRDbResourceSize.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.schema.size;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Schema objects related to this resource
```
Comment (lmsurpre) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/FHIRDbResourceSize.java:13]: Its the aggregated size for a resource type, not a single resource instance, right?  If so, I think adding this one word will help make that more clear:
```suggestion
 * Schema objects related to this resource type
```

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/ISizeReport.java (hunk 1 above)
Comment (punktilious) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/ISizeReport.java:16]: ok

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/PostgresSizeCollector.java (hunk 2 above)
Comment (punktilious) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/PostgresSizeCollector.java:88]: ok

File: fhir-persistence-schema/docs/SchemaToolUsageGuide.md (hunk 3 above)
Comment (punktilious) [reply fhir-persistence-schema/docs/SchemaToolUsageGuide.md:561]: good point. done.

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/FHIRDbResourceSize.java (hunk 4 above)
Comment (punktilious) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/FHIRDbResourceSize.java:13]: done

**Review by tbieste — APPROVED** [review]
Looks like issue #3284 was created to add a sort option to this, which I think is a good future enhancement.  Otherwise, LGTM.

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**